backend/src/Altinn.AccessManagement.UI/Altinn.AccessManagement.UI.Core/Models/SingleRight/CheckDelegationAccess/DelegationAccessCheckResponse.cs

```diff
@@ -48,7 +48,7 @@ public class DelegationAccessCheckResponse
         ///     The list of resources.
         /// </summary>
         [JsonProperty("resource")]
-        public List<Resource> Resource { get; init; }
+        public List<AttributeMatch> Resource { get; init; }
```

Comment:
Because it's an object created for this purpose so we don't have to create a new object everytime there's an object that has Id and Value. I'll rename it to plural form though. Also see Jon Kjetil's comment above, requesting me to use more already made objects, which we should aim to do as much as possible.

Reply:
![image](https://github.com/Altinn/altinn-access-management-frontend/assets/31471142/3352e14b-a236-4d59-9c06-2ae14265a166)

Reply:
I agree with the fact that we should try to re-use models whenever possible instead of making new ones all the time.
However, that only works if the model is named in a way that makes sense.

Right now, I pause in confusion every time I read that the Resource list is a list of AttributeMatches, cause unless I have misunderstood something (which ofc may be the case), that doesn't make sense. Then I remember that it is a reuse of a model and it isn't _really_ attribute matches, just an Id-value pair. When this thought process has to happen every time, it slows down readability and is downright misleading for someone new to the project.

I suggest that whenever a model is to be used and re-used as a generic one, it should also be named in a way that is generic enough to not confuse people. The name should then describe what it _is_ rather than its specific use case, since that use case can now vary. Like here, I would rename AttributeMatch to something like IdValuePair or something.

Reply:
Yeah renaming it seems like a good solution as my first thought too was that AttributeMatch was too abstract